sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/ShareDirectoryAsyncClient.java

```diff
@@ -161,9 +161,14 @@ public ShareFileAsyncClient getFileClient(String fileName) {
      * @return a ShareDirectoryAsyncClient that interacts with the specified directory
      */
     public ShareDirectoryAsyncClient getSubdirectoryClient(String subdirectoryName) {
-        String directoryPath = this.directoryPath + "/" + subdirectoryName;
-        return new ShareDirectoryAsyncClient(azureFileStorageClient, shareName, directoryPath, snapshot, accountName,
-            serviceVersion);
+        StringBuilder directoryPathBuilder = new StringBuilder()
```

Comment:
Should we also add this fix to getFileClient? So customers can get references to files in the root dir

Reply:
It looks like getFileClient already has support for root directory clients, though they have a bit of a simpler solution than a StringBuilder